app/models/ext_management_system.rb

```diff
@@ -101,6 +101,8 @@ def self.supported_for_create?
   validates :name,     :presence => true, :uniqueness => {:scope => [:tenant_id]}
   validates :hostname, :presence => true, :if => :hostname_required?
   validate :hostname_uniqueness_valid?, :hostname_format_valid?, :if => :hostname_required?
+  before_create :supported_for_create?
+  delegate :supported_for_create?, :to => :class
```

Comment:
Minor, but can you move this down a little?  In this commit, it's splitting the list of validate calls.